lib/jsdom/living/nodes/HTMLLabelElement-impl.js

```diff
@@ -29,6 +28,11 @@ function isLabelable(node) {
   return false;
 }
 
+function hasCommonAncestor(a, b) {
+  const ancestors = new Set(domSymbolTree.ancestorsIterator(a));
+  return domSymbolTree.ancestorsToArray(b).some(ancestor => ancestors.has(ancestor));
```

Comment:
Can you use `ancestorsIterator` as I originally proposed? That would obviate the need to walk through the entire ancestor chain for `b` if not needed.